File: Python/trees/level_order.py
import collections

from TreeNode import TreeNode
from typing import Optional
class Solution:

    def levelOrder(self,root: Optional[TreeNode]):
        q = collections.deque()
        q.append(root)
        result = []
        while q:
            level = []
            length = len(q)
            for i in range(length):
                node = q.popleft()
                if node:
                    level.append(node.val)
                    q.append(node.left)
                    q.append(node.right)
            if level:
                result.append(level)
        return result

File: Java/trees/level_order.java
import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class level_order {

    public List<List<Integer>> levelOrder(TreeNode root){
        Queue<TreeNode> q = new LinkedList<>();
        List<List<Integer>> result = new ArrayList<>();
        q.offer(root);
        while(!q.isEmpty()) {
            ArrayList<Integer> level = new ArrayList<>();
            int length = q.size();
            TreeNode node = q.poll();
            for(int i = 0; i < length; i++){
                if(node != null){
                    level.add(node.val);
                    q.offer(node.left);
                    q.offer(node.right);
                }
            }
            if(!level.isEmpty()){
                result.add(level);
            }
        }
        return result;
    }


}
